tasks/Module 3/task_1.py

```diff
@@ -0,0 +1,36 @@
+"""Sandwich calculator."""
+import pyinputplus as pyip
+
+BREAD_COST = {"wheat": 1, "white": 1.5, "sourdough": 1.5}
+MEAT_COST = {"chicken": 3, "turkey": 4, "ham": 3, "tofu": 4}
+CHEESE_COST = {"cheddar": 2, "swiss": 1.5, "mozzarella": 1}
+SAUSE_COST = {"mayo": 0.2, "mustard": 0.1, "lettuce": 0.2, "tomato": 0.1}
+
+
+def sandwich_maker():
+    """Build sandwich."""
+    print("Welcome to the sandwich ordering program!!! Please choose ingridients for your sandwich:")
+    print(' BREAD '.center(20, '='))
+    bread = pyip.inputMenu(list(BREAD_COST.keys()))
+    print(' MEAT '.center(20, '='))
+    meat = pyip.inputMenu(list(MEAT_COST.keys()))
+    if pyip.inputYesNo('Would you like a cheese? ') == 'yes':
+        cheese = pyip.inputMenu(list(CHEESE_COST.keys()))
+        cheese_to_print = f", {cheese}"
+    else:
+        cheese = ''
+        cheese_to_print = ''
+    if pyip.inputYesNo('Would you like a sause? ') == 'yes':
+        sause = pyip.inputMenu(list(SAUSE_COST.keys()))
+        sause_to_print = f", {sause}"
```

Comment:
I'm not a fan of hardcoding the logic to display the output of your program. I think this should be decided at the very end when actually printing the output. 

What would you say about creating a new dictionary, eg. `options` and append all the options user picked in there? eg. `options = {'wheat': 1, 'turkey': 3, ...}`

This way calculating costs and ingredients picked will be easier and more readable, and you don't need a special case for the cheese/no cheese, or sauce/no sauce.

Reply:
But it does not solve issue with formatting final message when optional parts are ordered and when not ordered. Does it?

Reply:
I think it does actually. 
You can just create a generic message `f"Your sandwich consists of: {''.join(options)}"`

So the idea behind it is to add the ingredient to `options` only if it was picked.

Reply:
perfect!